flashlight/app/lm/CMakeLists.txt

```diff
@@ -1,52 +1,27 @@
 cmake_minimum_required(VERSION 3.10)
 
-add_library(
-  flashlight-app-lm
-  ""
-  )
-
-target_sources(
-  flashlight-app-lm
+# ----------------------------- Binaries -----------------------------
+add_executable(fl_lm_train
+  ${CMAKE_CURRENT_LIST_DIR}/Train.cpp
+)
+target_sources(fl_lm_train
   PRIVATE
   ${CMAKE_CURRENT_LIST_DIR}/Trainer.cpp
-  )
-
-target_link_libraries(
-  flashlight-app-lm
-  PUBLIC
-  ${GLOG_LIBRARIES}
-  ${gflags_LIBRARIES}
-  flashlight
-  fl-libraries
-  flashlight-app-common
-  )
-
-target_include_directories(
-  flashlight-app-lm
-  PUBLIC
-  ${GLOG_INCLUDE_DIRS}
-  ${gflags_INCLUDE_DIRS}
-  )
-
-# ------------------------ Components ------------------------
-
-# Common
-include(${CMAKE_CURRENT_LIST_DIR}/common/CMakeLists.txt)
-
-# Data
-include(${CMAKE_CURRENT_LIST_DIR}/data/CMakeLists.txt)
-
-# ----------------------------- Binaries -----------------------------
-add_executable(fl_lm_train ${CMAKE_CURRENT_LIST_DIR}/Train.cpp)
+)
 add_executable(fl_lm_test ${CMAKE_CURRENT_LIST_DIR}/Test.cpp)
+
+target_sources(fl_lm_test
+  PRIVATE
+  ${CMAKE_CURRENT_LIST_DIR}/Trainer.cpp
+)
 add_executable(
   fl_lm_dictionary_builder
   ${CMAKE_CURRENT_LIST_DIR}/BuildDictionary.cpp
   )
 
-target_link_libraries(fl_lm_train flashlight-app-lm)
-target_link_libraries(fl_lm_test flashlight-app-lm)
-target_link_libraries(fl_lm_dictionary_builder flashlight-app-lm)
+target_link_libraries(fl_lm_train fl_pkg_text fl_pkg_runtime)
```

Comment:
As before, optional, but a CMake function like `fl_register_lm_binary` could do the following:
1. Add `Trainer.cpp` as a source
2. link to `fl_pkg_text fl_pkg_runtime`
3. Set the correct executable output directory